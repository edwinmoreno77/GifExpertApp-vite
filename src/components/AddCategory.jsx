import { useState } from "react"


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
    }
    return (
        <form onSubmit={e => onSubmit(e)}>
            <input type="text"
                placeholder="Search Gif"
                value={inputValue}
                onChange={e => handleInputChange(e)} // se deja visible el evento de forma educativa

            />
        </form>
    )
}
