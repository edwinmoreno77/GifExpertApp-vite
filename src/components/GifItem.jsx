import PropTypes from 'prop-types';


export const GifItem = ({ title, url, }) => {
    return (
        <div className="card animate__animated animate__heartBeat">
            <img src={url} alt={title} />
            <p className="p"><b>{title}</b></p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}