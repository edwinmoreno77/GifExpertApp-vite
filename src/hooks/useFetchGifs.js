import { useEffect, useState } from "react";
import { getGifs } from '../helpers/getGifs';



export const useFetchGifs = (cat) => {

    const [images, setImages] = useState([]);
    const [isloading, setIsLoading] = useState(true);

    const getImages = async () => {
        const newImages = await getGifs(cat);
        setImages(newImages);
        setIsLoading(false);
    }


    useEffect(() => {
        getImages();
    }, []);


    return {
        images: images,
        isLoading: isloading
    }

}
