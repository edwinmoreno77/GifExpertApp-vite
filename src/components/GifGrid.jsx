import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifItem } from './GifItem';





export const GifGrid = ({ cat }) => {

    const [images, setImages] = useState([]);

    const getImages = async () => {
        const newImages = await getGifs(cat);

        setImages(newImages);
    }

    useEffect(() => {
        getImages(cat);

    }, [])

    return (
        < >
            <h2>{cat}</h2>
            <div className='card-grid'>
                {
                    images.map((img) => (
                        <GifItem key={img.id} {...img} />
                    ))
                }
            </div>
        </>
    )
}