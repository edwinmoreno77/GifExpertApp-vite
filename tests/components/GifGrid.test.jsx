import { render, screen } from "@testing-library/react";
import { GifGrid } from "../../src/components/GifGrid";
import { useFetchGifs } from '../../src/hooks/useFetchGifs';

jest.mock('../../src/hooks/useFetchGifs');

describe('<GifGrid/> test ', () => {
    const cat = 'Goku';

    test('should render <GifGrid/>', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });

        const { container } = render(<GifGrid cat={cat} />);
        expect(container).toMatchSnapshot();

    })

    test('should render <GifGrid/> with loading and category', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });

        render(<GifGrid cat={cat} />);
        screen.getByText('Loading...');
        screen.getByText(cat);
        // screen.debug();

    })

    test('should display items when images are loaded from useFetch', () => {
        useFetchGifs.mockReturnValue({
            images: [
                {
                    id: '123',
                    title: 'Goku',
                    url: 'https://media1.giphy.com/media/3o7btLq1Qz7qQqQq2U/giphy.gif'
                },
                {
                    id: '456',
                    title: 'Vegeta',
                    url: 'https://media1.giphy.com/media/3o7btLq1Qz7qQqQq2U/giphy.gif'
                }
            ],

            isLoading: false,
        });

        render(<GifGrid cat={cat} />);
        // screen.debug();
        expect(screen.getAllByRole('img').length).toBe(2);

    })
})