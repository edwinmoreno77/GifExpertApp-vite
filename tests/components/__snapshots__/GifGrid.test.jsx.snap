// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid/> test  should render <GifGrid/> 1`] = `
<div>
  <h2
    class="animate__animated animate__fadeIn"
  >
    Goku
  </h2>
  <h2
    class="animate__animated animate__flash"
  >
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
